src/pages/List.js

```diff
@@ -77,27 +136,125 @@ export default function List({ token }) {
       }
     });
   }
+
+  const alphabetizeListItems = (list) => {
+    const sortedList = list.sort((a, b) => {
+      if (a.data().item_name.toLowerCase() < b.data().item_name.toLowerCase()) {
+        return -1;
+      }
+      if (a.data().item_name.toLowerCase() > b.data().item_name.toLowerCase()) {
+        return 1;
+      }
+      return 0;
+    });
+    return sortedList;
+  };
+
+  const filterByUserInput = (item) => {
+    // first alphabetize all the items then filter for the user's searched item(s)
+    return alphabetizeListItems(item.docs).filter(
+      (doc) =>
+        doc
+          .data()
+          .item_name.toLowerCase()
+          .includes(query.toLowerCase().trim()) || query === '',
+    );
+  };
+
+  const filterByLessThanSevenDays = (listItems) => {
+    // filter the items by user input or render all items if no input
+    const alphabetizedUserInputOrAllItems = filterByUserInput(listItems);
+
+    // filter into the green category of less than 7 days
+    return alphabetizedUserInputOrAllItems.filter((item) => {
+      if (item.data().times_purchased === 0) {
+        return item.data().purchase_frequency === 7;
+      } else {
+        return item.data().last_estimate < 7 && !checkForInactiveItem(item);
+      }
+    });
+  };
+
+  const filterByMoreThanSevenDaysAndLessThanThirtyDays = (listItems) => {
+    // filter the items by user input or render all items if no input
+    const alphabetizedUserInputOrAllItems = filterByUserInput(listItems);
+
+    // filter items into the purple category of more than 7 days and less than 30 days
+    return alphabetizedUserInputOrAllItems.filter((item) => {
+      if (item.data().times_purchased === 0) {
+        return item.data().purchase_frequency === 14;
+      } else {
+        return (
+          item.data().last_estimate >= 7 &&
+          item.data().last_estimate <= 30 &&
+          !checkForInactiveItem(item)
+        );
+      }
+    });
+  };
+
+  const filterByMoreThanThirtyDays = (listItems) => {
+    // filter the items by user input or render all items if no input
+    const alphabetizedUserInputOrAllItems = filterByUserInput(listItems);
+
+    // filter items into the red category of more than 30 days
+    return alphabetizedUserInputOrAllItems.filter((item) => {
+      if (item.data().times_purchased === 0) {
+        return item.data().purchase_frequency === 30;
+      } else {
+        return item.data().last_estimate > 30 && !checkForInactiveItem(item);
+      }
+    });
+  };
```

Comment:
This duplication of efforts is not ideal but per a discussion with @jamesncox this will be left for future improvements.